docs/advanced-configurations.rst

```diff
@@ -304,3 +304,118 @@ If self-signed certificates are used for an HTTPS based internal repository, a C
+
+-------------------
+Dockerfile Overview
+-------------------
+
+To build the precompiled container, the Dockerfile is constructed in a multistage fashion.
+This approach is used to optimize the resulting container image size and reduce the number of dependencies included in the final image.
+
+The Dockerfile consists of the following stages:
+
+1. **Base Image Update**: The base image is updated and common requirements are installed. This stage sets up the basic environment for the subsequent stages.
+
+2. **Download Driver Sources**: This stage downloads the Mellanox OFED driver sources to the specified path. It prepares the necessary files for the driver build process.
+
+3. **Build Driver**: The driver is built using the downloaded sources and installed on the container. This stage ensures that the driver is compiled and configured correctly for the target system.
+
+4. **Install precompiled driver**: Finally, the precompiled driver is installed on clean container. This stage sets up the environment to run the NVIDIA NIC drivers on the target system.
+
+
+---------------------------------
+Common mandatory build parameters
+---------------------------------
+
+Before building the container, you need to provide following parameters as `build-arg` for container build:
+
+1. `D_OS`: The Linux distribution (e.g., ubuntu22.04 / rhel9.2)
+2. `D_ARCH`: Compiled Architecture
+3. `D_BASE_IMAGE`: Base container image
+4. `D_KERNEL_VER`: The target kernel version (e.g., 5.15.0-25-generic / 5.14.0-284.32.1.el9_2.x86_64)
+5. `D_OFED_VERSION`: NVIDIA NIC drivers version (e.g., 24.01-0.3.3.1)
+
+**NOTE:** Check desired NVIDIA NIC drivers sources[^1] availability for designated container OS, only versions available on download page can be utilized
+
+------------------------------
+RHEL-specific build parameters
+------------------------------
+
+1. `D_BASE_IMAGE`: DriverToolKit container image
+
+**NOTE:** DTK (DriverToolKit) is tightly coupled with specific kernel versions, verify match between kernel version to compile drivers for, versus DTK image.
+
+2. `D_FINAL_BASE_IMAGE`: Final container image, to install compiled driver
+
+**NOTE:** For proper Network Operator functionality container tag name must be in following pattern: **driver_ver-container_ver-kernel_ver-os-arch**. For example: 24.01-0.3.3.1-0-5.15.0-25-generic-ubuntu22.04-amd64
+
+~~~~~~~~~~~~
+RHEL example
+~~~~~~~~~~~~
+
+To build RHEL-based image please use provided :download:`Dockerfile <files/RHEL_Dockerfile>` and :download:`entrypoint script <files/entrypoint.sh>`.
+
+.. code-block:: bash
+
+   podman build \
+    --build-arg D_OS=rhel9.2 \
+    --build-arg D_ARCH=x86_64 \
+    --build-arg D_KERNEL_VER=5.14.0-284.32.1.el9_2.x86_64 \
+    --build-arg D_OFED_VERSION=24.01-0.3.3.1 \
+    --build-arg D_BASE_IMAGE="registry.redhat.io/openshift4/driver-toolkit-rhel9:v4.13.0-202309112001.p0.gd719bdc.assembly.stream" \
```

Comment:
D_BASE_IMAGE is the DTK image should explain above to use this param for OCP4.14^ and how to detect required DTK image

Reply:
done